content-script.js

```diff
@@ -433,6 +546,112 @@ const timer = setInterval(() => {
   if (ytShorts) checkVolume(ytShorts);
 }, 100);
 
+/**
+ * Converts a formatted number to its full integer value.
+ * @param {string} string value to be converted (eg: 1.4M, 1,291 or 727) 
+ * @returns converted number
+ */
+function convertLocaleNumber( string )
+{
+  if ( typeof string !== "string" ) return
+  
+  // todo  - add formats from other langs
+  const multipliers = {
+      // English
+	  "b":   1_000_000_000,
+	  "m":   1_000_000,
+	  "k":   1_000,
+
+	  // Italian
+	  "mln": 1_000_000,
+
+	  // Indian English
+	  "lakh": 100_000,
+
+	  // Portuguese
+	  "mil": 1_000,
+
+	  // Spanish
+	  "mil": 1_000,
+
+	  // French
+	  "mio": 1_000_000,
+	  "md":  1_000,
+
+	  // German
+	  "mio": 1_000_000,
+	  "mrd": 1_000_000_000,
+	  "tsd": 1_000,
+
+	  // Japanese
+	  "億":  1_000_000_000,
+	  "万":  10_000,
+
+	  // Chinese (Simplified)
+	  "亿":  1_000_000_000,
+	  "万":  10_000,
+
+	  // Chinese (Traditional)
+	  "億":  1_000_000_000,
+	  "萬":  10_000,
+
+	  // Russian
+	  "млн": 1_000_000,
+	  "тыс": 1_000,
+
+	  // Hindi
+	  "करोड़": 10_000_000,
+	  "लाख":  100_000,
+
+	  // Arabic
+	  "مليون":   1_000_000,
+	  "مليار":   1_000_000_000,
+	  "ألف":     1_000,
+
+	  // Korean
+	  "억":  100_000_000,
+	  "만":  10_000,
+
+	  // Turkish
+	  "milyon":    1_000_000,
+	  "milyar":    1_000_000_000,
+	  "bin":       1_000,
+
+	  // Vietnamese
+	  "triệu":    1_000_000,
+	  "tỷ":       1_000_000_000,
+	  "nghìn":    1_000,
+
+	  // Thai
+	  "ล้าน":    1_000_000,
+	  "พันล้าน": 1_000_000_000,
+	  "พัน":     1_000,
+
+	  // Dutch
+	  "mio":  1_000_000,
+	  "mld":  1_000_000_000,
+	  "k":    1_000,
+
+	  // Greek
+	  "εκ":   1_000_000,
+	  "δισ":  1_000_000_000,
+	  "χιλ":  1_000,
+
+	  // Swedish
+	  "mn":   1_000_000,
+	  "md":   1_000_000_000,
+	  "t":    1_000,
+  }
+  const end = string.length - 1
+  const multiplier = string.toLowerCase().replace( /[^a-z]/g, "" )
+  const hasMultiplier = Object.keys( multipliers ).includes( multiplier )
+
+  if ( hasMultiplier )
+    return +string.slice( 0, end ).replace( /,|\./g, "" ) * multipliers[ multiplier ]
+
+  return +string.slice( 0, end + 1 ).replace( /,|\./g, "" ) 
```

Comment:
Wouldn't this cause inaccuracies in the final value? Namely if the like count is `5.6M` and the if the dot is removed it will be like multiplied by ten here.

Reply:
It would, replacing 
`string.slice(0, end) `

with 

`string.slice(0, end + 1)`

should fix it.

Reply:
nevermind that won't fix it, it'd actually return 0

Reply:
Fixed this issue here
https://github.com/adsuth/better-yt-shorts/pull/2#issue-1816827392